lxd/storage/drivers/driver_powerflex_utils.go

```diff
@@ -931,19 +914,26 @@ func (d *powerflex) mapVolume(vol Volume) (revert.Hook, error) {
 		reverter.Add(func() { _ = client.deleteHostVolumeMapping(hostID, volumeID) })
 	}
 
-	if d.config["powerflex.mode"] == powerFlexModeNVMe {
-		// Connect to the NVMe/TCP subsystem.
-		// We have to connect after the first mapping was established.
-		// PowerFlex does not offer any discovery log entries until a volume gets mapped to the host.
-		// This action is idempotent.
-		cleanup, err := d.connectNVMeSubsys()
-		if err != nil {
-			return nil, err
-		}
+	pool, err := d.resolvePool()
+	if err != nil {
+		return nil, err
+	}
 
-		reverter.Add(cleanup)
+	domain, err := d.client().getProtectionDomain(pool.ProtectionDomainID)
+	if err != nil {
+		return nil, err
+	}
+
+	targetQN := domain.SystemID
+	targetAddr := d.config["powerflex.sdt"]
+
+	err = d.connector().Connect(d.state.ShutdownCtx, targetAddr, targetQN)
+	if err != nil {
+		return nil, err
 	}
 
+	reverter.Add(func() { _ = d.connector().Disconnect(targetQN) })
```

Comment:
`DisconnectAll`?

Reply:
I would avoid DisconnectAll, as previously discussed, this would disconnect all NVMe volumes from the LXD host, which is not desired.